### ISSUE TITLE
[TEST-FAILURE] XmlConfigImportVariableReplacementTest.testThreeResourceCyclicImportThrowsException

### ISSUE THREAD

**Issue by bilalyasar** [hazelcast/hazelcast#4846]
```
java.lang.Exception: Unexpected exception, expected<com.hazelcast.config.InvalidConfigurationException> but was<com.hazelcast.core.HazelcastException>
	at com.sun.org.apache.xerces.internal.parsers.DOMParser.parse(DOMParser.java:257)
	at com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:347)
	at javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:121)
	at com.hazelcast.config.XmlConfigBuilder.parse(XmlConfigBuilder.java:210)
	at com.hazelcast.config.AbstractConfigBuilder.loadAndReplaceImportElement(AbstractConfigBuilder.java:95)
	at com.hazelcast.config.AbstractConfigBuilder.replaceImportElementsWithActualFileContents(AbstractConfigBuilder.java:80)
	at com.hazelcast.config.AbstractConfigBuilder.process(AbstractConfigBuilder.java:65)
	at com.hazelcast.config.XmlConfigBuilder.parseAndBuildConfig(XmlConfigBuilder.java:200)
	at com.hazelcast.config.XmlConfigBuilder.build(XmlConfigBuilder.java:184)
	at com.hazelcast.config.XmlConfigBuilder.build(XmlConfigBuilder.java:177)
	at com.hazelcast.config.XmlConfigImportVariableReplacementTest.buildConfig(XmlConfigImportVariableReplacementTest.java:489)
	at com.hazelcast.config.XmlConfigImportVariableReplacementTest.testThreeResourceCyclicImportThrowsException(XmlConfigImportVariableReplacementTest.java:170)
```
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x/2884/com.hazelcast$hazelcast/testReport/com.hazelcast.config/XmlConfigImportVariableReplacementTest/testThreeResourceCyclicImportThrowsException/

**Comment by pveentjer** [comment]
Should InvalidConfigurationException not be a subclass of HazelcastException?

**Comment by bilalyasar** [comment]
yes, it doesn't have to subclass of HazelcastException.

**Comment by pveentjer** [comment]
Most exception in HZ extend HazelcastException.

**Comment by mesutcelik** [comment]
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-OracleJDK1.6/478/com.hazelcast$hazelcast/testReport/com.hazelcast.config/XmlConfigImportVariableReplacementTest/testThreeResourceCyclicImportThrowsException/

**Comment by mesutcelik** [comment]
https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-OracleJDK1.7/60/com.hazelcast$hazelcast/testReport/com.hazelcast.config/XmlConfigImportVariableReplacementTest/testThreeResourceCyclicImportThrowsException/

**Comment by bilalyasar** [comment]
closed via #4851
